inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -0,0 +1,135 @@
+        penalty, to be applied in the student's final grade for the
+        respective task.
+    """
+
+    def __init__(self, username, task_id, database):
+
+        self._username = username
+        self._task_id = task_id
+        self._database = database
+
+    def get_hint_content_by_status(self, hints):
+
+        """
+            This is a method to check each hint unlocked status, and return the left content
+            if it is unlocked. Also set a 'True' value in the 'unlocked'
+            attribute for the unlocked hints to show their additional data
+            in the modal hints in the task.
+        """
+        user_hints = self._database.user_hints.find_one({"taskid": self._task_id,
+                                                         "username": self._username})
+        if user_hints is None:
+            user_hints = self.insert_default_user_hints()
+            return 200, ""
+
+        self.update_unlocked_user_hints(user_hints, hints)
+
+        hints_to_show = {}
+        unlocked_hints = user_hints["unlocked_hints"]
+        unlocked_hints_data = {}
+        for hint in unlocked_hints:
+            unlocked_hints_data[hint["id"]] = hint["penalty"]
+
+        for key, hint in hints.items():
+            unlocked_hints_content = {
+                "title": hint["title"],
+                "content": None,
+                "penalty": hint["penalty"],
+                "unlocked": False,
+            }
+            if hint["id"] in unlocked_hints_data.keys():
+                parsed_hint_content = self.parse_rst_content(hint["content"])
+                unlocked_hints_content["content"] = parsed_hint_content
+                unlocked_hints_content["unlocked"] = True
+                unlocked_hints_content["penalty"] = unlocked_hints_data[hint["id"]]
+
+            hints_to_show[key] = unlocked_hints_content
+
+        return hints_to_show
+
+    def update_unlocked_user_hints(self, user_hints, hints):
+
+        """ Method to compare the user hints with the task hints, and update them"""
+        unlocked_hints = user_hints["unlocked_hints"]
+        hints_ids = [hint["id"] for hint in hints.values()]
+        for hint in unlocked_hints:
+            if not hint["id"] in hints_ids:
+                self.remove_unlocked_hint(hint)
+
+        self.update_total_penalty()
+
+    def insert_default_user_hints(self):
+
+        user_hints = self._database.user_hints.insert(
+            {"taskid": self._task_id, "username": self._username, "unlocked_hints": [], "penalty": 0})
+        return user_hints
+
+    def is_hint_unlocked(self, hint_id):
+
+        """ Method to check if the hint is already in the user hints """
+        user_hints = self._database.user_hints.find_one({"taskid": self._task_id, "username": self._username})
+        unlocked_hints = user_hints["unlocked_hints"]
+
+        for hint in unlocked_hints:
+            if hint_id is hint["id"]:
```

Comment:
```suggestion
            if hint_id == hint["id"]:
```
The `is` operator checks that both are the same instance I think.